code/script.js

```diff
@@ -256,70 +269,111 @@ const checkQuestion = () => {
+     if (keep) {
+      alert(`Yes, the person is a ${value}! Keep all people that are ${value}`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+    } else {
+      alert(`No, the person is not a  ${value}! Remove all people that's a ${value}`)
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+
+  } else if (category === 'hair') {
+    if (keep) {
+      alert( `Yes, the person has ${value} hair! Keep all people with ${value} hair.` )
+      //for hair and eyes 
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value) 
+    } else {
+      alert(`No, the person does not have ${value} hair! Remove all people with ${value} hair.` )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+  
   } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+       // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert( `Yes, the person has ${value} eyes! Keep all people with ${value} eyes.` )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
     } else {
       // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(`No, the person does not have ${value} eyes! Remove all people with ${value} eyes.` )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
   // Invoke a function to redraw the board with the remaining people.
+  generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
+// store the interaction from the player in a variable.
+// remember the confirm() ?
+// If the player wants to guess, invoke the checkMyGuess function.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const playerGuess = confirm(`Are you sure you want to guess on ${personToConfirm}?`)
+  if (playerGuess) {
+    checkMyGuess(personToConfirm)
+  } else {
+    alert('Got it, keep playing!')
+  }
 }
```

Comment:
Smart to have an alert if the person clicks cancel on the confirm. User friendly! :)